src/assessments/seo/introductionKeywordAssessment.js

```diff
@@ -1,62 +1,99 @@
-var AssessmentResult = require( "../../values/AssessmentResult.js" );
+const Assessment = require( "../../assessment.js" );
+const AssessmentResult = require( "../../values/AssessmentResult.js" );
+const merge = require( "lodash/merge" );
 
 /**
- * Returns a score and text based on the firstParagraph object.
- *
- * @param {object} firstParagraphMatches The object with all firstParagraphMatches.
- * @param {object} i18n The object used for translations
- * @returns {object} resultObject with score and text
+ * Assessment to check whether the keyphrase is encountered in the first paragraph of the article.
  */
-var calculateFirstParagraphResult = function( firstParagraphMatches, i18n ) {
-	const url = "<a href='https://yoa.st/2pc' target='_blank'>";
+class IntroductionHasKeywordAssessment extends Assessment {
+	/**
+	 * Sets the identifier and the config.
+	 *
+	 * @param {Object} config The configuration to use.
+	 *
+	 * @returns {void}
+	 */
+	constructor( config = {} ) {
+		super();
+
+		let defaultConfig = {
+			parameters: {
+				recommendedMinimum: 1,
+			},
+			scores: {
+				good: 9,
+				bad: 3,
+			},
+			url: "<a href='https://yoa.st/2pc' target='_blank'>",
+		};
+
+		this.identifier = "introductionKeyword";
+		this._config = merge( defaultConfig, config );
+	}
+
+	/**
+	 * Assesses the presence of keyphrase in the first paragraph.
+	 *
+	 * @param {Paper} paper The paper to use for the assessment.
+	 * @param {Researcher} researcher The researcher used for calling research.
+	 * @param {Jed} i18n The object used for translations.
+	 *
+	 * @returns {AssessmentResult} The result of this assessment.
+	 */
+	getResult( paper, researcher, i18n ) {
```

Comment:
I'm not sure how exactly the i18n object works here. If you have an idea how to tackle this I'd definitely be open to changing that.

Reply:
I think the same applies as with `getResearchResults`. Out of scope of this refactor. And maybe ask @atimmer how we want to be able to use getResult. For me it comes down to:
- Do we need the same results in different languages? Then it makes sense to pass along the i18n. ( And then I could still argue that the assessment should only return scores and not sentences ).

Looking at the code it does not look like we support this, as there is only one `i18n` which comes from the `app.js` through the Assessors. And it holds only the current language.

Therefor it does seem easier to me to just pass it along to the constructor instead of every getResult call. But pretty minor and not 100% clear, so let's not 👍